app/views/posts/_post.html.haml

```diff
@@ -20,12 +20,26 @@
             - if post.image.attached?
               = image_tag (post.image), alt: 'post image', class: 'post-image'
     .icons#icons-post
-      %svg{class: 'bi bi-chat-left show-comments', 'data-comments': "comments-post-#{post.id}", width: '28', height: '28', fill: 'currentColor', viewBox: '0 0 16 16', xmlns: 'http://www.w3.org/2000/svg'}
-        %path{d: 'M14 1a1 1 0 0 1 1 1v8a1 1 0 0 1-1 1H4.414A2 2 0 0 0 3 11.586l-2 2V2a1 1 0 0 1 1-1zM2 0a2 2 0 0 0-2 2v12.793a.5.5 0 0 0 .854.353l2.853-2.853A1 1 0 0 1 4.414 12H14a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2z'}
-      - if current_user.likes.find_by(likeable_id: post.id)
-        = render partial: 'likes/unlike', locals: { likeable: post }
-      - else
-        = render partial: 'likes/like', locals: { likeable: post }
+      - if post.user == current_user && controller_name == 'posts' && action_name == 'show'
```

Comment:
I'm almost sure you can reduce the check of controller or method with something like `current_page?(aca el nombre del path)`